Update min supported version numbers

reykjalin opened the pull request:
Also updates the PHPCS sniffer rules.


## Changes proposed in this Pull Request:

- Update minimum supported PHP version to version 7.0.
    - This is a sort of transitive dependency because we require at least WC 5.6 which requires at least PHP 7.0.
- Update the PHPCS php version sniff to version 7.0.
- Update minimum supported WP version in readme.txt to 5.6 instead of 4.6.
- Update some out of date version constants.

## Testing instructions

1. Try using the null-coalescing operator in a code editor.
    - For example; `$foo = 'bar' ?? 'baz';`
2. Make sure phpcs does not mark that as an issue.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry (or does not apply)
-   [x] Tested on mobile (or does not apply)

Commit:
Update min supported version numbers

Also updates the PHPCS sniffer rules.

Commit:
add changelog entry

diegocurbelo committed:
Merge branch 'develop' into fix/update-min-php-to-version-7

diegocurbelo committed:
Hide notice for 'last version compatible with WC..' when testing Stripe notices

diegocurbelo reviewed (approved):
Looks good! 🚢 🚢 

_I fixed the test that was failing after I merged https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2095 into develop 😭_

Commit:
Revert "Hide notice for 'last version compatible with WC..' when testing Stripe notices"

This reverts commit 343563d74c0cb2e352dd433dadb02ac41e760088.

Commit:
Take version dependent notice into account in test

This fixes an issue where unit tests were failing while testing certain
versions of WooCommerce.

reykjalin commented:
@diegocurbelo I think the [approach I took to fix the tests](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2109/commits/b646d5afc26aff0e72527db1ce11d603cc35a7b7) is a little bit more robust. Let me know what you think 🙂 

I took the liberty of reverting your changes first so the GitHub checks should be accurate 👍 

**EDIT** took a bit more than the commit I linked to above — my local setup had reverted to the latest version of WC instead of the failing one which is why I was under the impression everything was working 😅 . I've fixed the issues so the tests now take into account the possibility of an additional notice in some circumstances 👍

Commit:
Fix expected notices for failing tests

Commit:
Fix extra notice check in tests

Commit:
fix upe notice tests

Commit:
Merge branch 'develop' into fix/update-min-php-to-version-7
